### PR TITLE
Add benchmark results for 1TB training and CCFRAUD finetune, bugs fixes

### PR DESCRIPTION
**Pull request by ww-aims**
## Purpose

* Add 1TB training for pneumonia with DDP and finetune results for CCFRAUD
* Modify hyper-parameters and architecture for CCFRAUD model
* Bug fix for multi-nodes DDP training
* Bug fix for EDA component

## What is the expected review turnaround time?

Urgency:
- [ ] High (needs review today)
- [x] Medium (needs review within a few days - most common)
- [ ] Low (can wait a week)

## Does this introduce a breaking change?

```
[ ] Yes
[x]  No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: examples/pipelines/ccfraud/config.yaml [hunk 1]
```diff
@@ -57,15 +57,15 @@ data_analysis_parameters:
...
+  epochs: 10
   lr: 1e-3
-  batch_size: 500
+  batch_size: 1000
   # Property model_name defines model to be used for training and 
   # must be one of the models defined in the traininsilo/model.py 
   # file for the CCFRAUD component. Options include: 
   # SimpleLinear, SimpleLSTM, SimpleVAE
-  model_name: SimpleLinear
+  model_name: SimpleLSTM
```
Comment (majercakdavid) [inline examples/pipelines/ccfraud/config.yaml:68]: NIT: Why to use LSTM over Linear?

File: examples/pipelines/ccfraud/submit.py [hunk 2]
```diff
@@ -241,7 +241,7 @@ def fl_ccfraud_basic():
                     mode=silo_config.testing_data.mode,
                     path=silo_config.testing_data.path + f"/test.csv",
                 ),
-                metrics_prefix=silo_config.compute,
+                metrics_prefix=silo_config.computes[0],
```
Comment (majercakdavid) [inline examples/pipelines/ccfraud/submit.py:244]: @garg-amit when did we change schema to include multiple computes and why is this necessary?

File: examples/components/CCFRAUD/traininsilo/models.py [hunk 3]
```diff
@@ -67,7 +67,7 @@ def __init__(self, input_dim) -> None:
         self.lstm = nn.LSTM(
             input_size=input_dim,
             hidden_size=128,
-            num_layers=4,
+            num_layers=1,
```
Comment (majercakdavid) [inline examples/components/CCFRAUD/traininsilo/models.py:70]: NIT: What is the gist behind changing number of layers to 1?

File: examples/pipelines/ccfraud/config.yaml (hunk 1 above)
Comment (ww-aims) [reply examples/pipelines/ccfraud/config.yaml:68]: This is because LSTM performs better than Linear for this dataset. All the hyper-params and model architectures are tuned for better performance.

File: examples/components/CCFRAUD/traininsilo/models.py (hunk 3 above)
Comment (ww-aims) [reply examples/components/CCFRAUD/traininsilo/models.py:70]: It's for the overall performance, the same as mentioned above.

File: examples/pipelines/ccfraud/submit.py (hunk 2 above)
Comment (ww-aims) [reply examples/pipelines/ccfraud/submit.py:244]: in our config, we do not have "compute" anymore but "computes" for each silo. This change simply avoids the error report

File: docs/concepts/benchmarking.md [hunk 4]
```diff
@@ -81,9 +81,9 @@ For ner, FL takes only 1.3% longer wall time than centralized-1/3, and only 0.1%
 
 #### CCFRAUD
 <p align="center">
-    <img src="../pics/cc_time.jpg" alt="ccfraud training time" width="600"/>
+    <img src="../pics/ccfraud_time.jpg" alt="ccfraud training time" width="600"/>
```
[the remark this answers is not in the document]
Comment (ww-aims) [reply docs/concepts/benchmarking.md:84]: Yes that's right. In the ccfraud case it is the sum of all gpus

File: examples/components/CCFRAUD/traininsilo/run.py [hunk 5]
```diff
@@ -162,10 +163,18 @@ def __init__(
             self.train_sampler_ = None
             self.test_sampler_ = None
 
+        # get number of cpu to load data for each gpu
+        num_workers_per_gpu = int(
+            multiprocessing.cpu_count() // int(os.environ["WORLD_SIZE"])
```
Comment (garg-amit) [inline examples/components/CCFRAUD/traininsilo/run.py:168]: let's use `os.environ.get("WORLD_SIZE", "1")` instead of os.environ["WORLD_SIZE"] to avoid any errors when running locally.

File: examples/components/CCFRAUD/traininsilo/run.py [hunk 6]
```diff
@@ -599,7 +608,7 @@ def run(args):
         total_num_of_iterations=args.total_num_of_iterations,
         experiment_name=args.metrics_prefix,
         iteration_name=args.iteration_name,
-        device_id=int(os.environ["RANK"]),
+        device_id=int(os.environ["LOCAL_RANK"]),
```
Comment (garg-amit) [inline examples/components/CCFRAUD/traininsilo/run.py:611]: similarily here and also in other scripts.

File: examples/pipelines/ccfraud/submit.py (hunk 2 above)
Comment (garg-amit) [reply examples/pipelines/ccfraud/submit.py:244]: The multiple computes changes were introduced in the Jan's release https://github.com/Azure-Samples/azure-ml-federated-learning/releases/tag/v0.4.0

File: docs/concepts/benchmarking.md [hunk 7]
```diff
@@ -108,18 +108,27 @@ For pneumonia, FL achieves higher accuracy than centralized-1/3, while slightly
...
+Scalability is critical for industry applications of FL on large datasets. One benefit of using FL on Azure ML is that it supports distributed training (multi GPUs and multi nodes). For this reason, we support distributed training for each real-world example, empowered by Pytorch Distributed Data Parallel (DDP) module. To test the scalability of our implementation, we artifically replicated each datasets by 10 times, and record the training time per epoch for each silo when such data is trained on different number of GPUs and machines. For pneumonia example, we also expanded the data to 1TB to demonstrate the scalability of our implementation in extreme cases.
```
Comment (jfomhover) [inline docs/concepts/benchmarking.md:119]: typo: artifically

File: docs/concepts/benchmarking.md [hunk 8]
```diff
@@ -81,9 +81,9 @@ For ner, FL takes only 1.3% longer wall time than centralized-1/3, and only 0.1%
 
 #### CCFRAUD
 <p align="center">
-    <img src="../pics/cc_time.jpg" alt="ccfraud training time" width="600"/>
```
Comment (jfomhover) [inline docs/concepts/benchmarking.md:84]: don't forget to delete the old files, they don't seem to have been deleted from the PR?

File: examples/pipelines/ccfraud/submit.py (hunk 2 above)
Comment (jfomhover) [reply examples/pipelines/ccfraud/submit.py:244]: the goal here was to have cpu+gpu for each pipeline, cpu for processing, gpu for training

**Review by garg-amit — APPROVED** [review]

**Review by jfomhover — APPROVED** [review]
